annotator/resolved_keyword_annotator.py

```
@@ -46,21 +49,36 @@ def annotate(self, doc):
 
         ngrams = list(set(span_text_to_spans.keys()))
         cursor = self.connection.cursor()
-        results  = cursor.execute('''
-        SELECT *
-        FROM synonyms
-        JOIN entity_labels USING ( uri )
-        WHERE synonym IN (''' +
-        ','.join('?' for x in ngrams) +
-        ')', ngrams)
 
         spans_to_resolved_keywords = defaultdict(list)
+        uris = set()
+        ordered_ngram_iter = iter(sorted(ngrams))
+        try:
+            ngram = next(ordered_ngram_iter)
+            for result in cursor.execute('SELECT * FROM synonyms ORDER BY synonym'):
+                while ngram < result['synonym']:
+                    ngram = next(ordered_ngram_iter)
+                if ngram == result['synonym']:
+                    for span in span_text_to_spans[ngram]:
+                        spans_to_resolved_keywords[span].append(result)
+                        uris.add(result['uri'])
+        except StopIteration:
+            pass
```

Just want to confirm that you want to eat this exception?